pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -584,19 +599,28 @@ func (vs *VSphere) DisksAreAttached(volPaths []string, nodeName k8stypes.NodeNam
 		}
 		vm, err := vs.getVMByName(ctx, nodeName)
 		if err != nil {
+			if vclib.IsNotFound(err) {
+				glog.Infof("Node %q does not exist, all the disks on this node will be detached automatically.", nodeName)
+				// make all the disks as detached and return false without error.
+				attached := make(map[string]bool)
+				for _, volPath := range volPaths {
+					attached[volPath] = false
+				}
+				return attached, nil
+			}
 			glog.Errorf("Failed to get VM object for node: %q. err: +%v", vSphereInstance, err)
 			return nil, err
 		}
-		nodeExist, err := vm.Exists(ctx)
+		isActive, err := vm.IsActive(ctx)
 		if err != nil {
-			glog.Errorf("Failed to check whether node %q exist. err: %+v", vSphereInstance, err)
+			glog.Errorf("Failed to check whether node %q is Active. err: %+v", vSphereInstance, err)
 			return nil, err
 		}
-		if !nodeExist {
-			glog.Errorf("DisksAreAttached failed to determine whether disks %v are still attached: node %q does not exist",
+		if !isActive {
+			glog.Errorf("DisksAreAttached failed to determine whether disks %v are still attached: node %q is powered off",
 				volPaths,
 				vSphereInstance)
-			return nil, fmt.Errorf("DisksAreAttached failed to determine whether disks %v are still attached: node %q does not exist",
+			return nil, fmt.Errorf("DisksAreAttached failed to determine whether disks %v are still attached: node %q is powered off",
```

Comment:
Returned error should be different. When the Node VM is powered off, we can still determine whether disks are attached to powered off VM or not. Current error message does not look good in this case.